canonical-web-and-design/maas-ui
# feat(ui): create OS formatting hook for use in machine list + details

## Done

- Added a hook `useFormattedOS` which formats a machine's OS and release into human readable text, e.g. `"ubuntu/focal"` => `"Ubuntu 20.04 LTS"`
- Updated `getStatusText()` util to be separate from OS formatting
- Converted some files to TS

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check the machine details page of a deploying or deployed machine and see that the OS shows under the status
- Check that the machine list still shows the formatted OS too

## Fixes

Fixes #1867 

## Screenshot
![Screenshot_2020-11-12 clymer maas details björnstack MAAS](https://user-images.githubusercontent.com/25733845/98902383-5d925c00-2501-11eb-95c9-de552ad83321.png)
